src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -286,6 +287,7 @@ public void WriteException(BinaryAnalyzerContext context, string errorMessage)
                 { "language", string.Empty },
                 { "debuggingFileName", string.Empty },
                 { "debuggingGuid", string.Empty },
+                 { "debuggingFileLastModifiedDateUTC", DateTime.MaxValue.ToString() },
```

Comment:
The reason I kept the default value to the maximum timestamp is as follows: the reason the last modified date was requested was to compare it with the date of the last build, and reject the dll if the pdb is stale. With a maximum value, we indicate the scenario that the date could not be determined, and the alert corresponding to the dll should be factored for mitigation.

Reply:
agree with the rationale here.

though I lean more towards:
here writes string.Empty, 
and above check if it is Datetime default then writes string.Empty
The user of the data do the checking if empty.

The benefit will be consistency, and easier to read.
anything empty is no data.
e.g.
`"C:\file.exe",,,,,,,,,`
compare to 
`"C:\file.exe",,,,"12/31/9999 11:59:59 PM",,,,,`

Reply:
ah I think of an possible issue, the class `CompilerData`and the `context.CompilerDataLogger.Write()` is used both in PE 4001 and non-PE 4002

I think we intent to change PE only, 
can you debug and see if 4002 is till working good?
I guess without any change, 4002 will write DateTime.Min because it is the default of not nullable DateTime type

Reply:
@shaopeng-gh OK that makes sense. Here's the change I made:
1. Switched the type of `DebuggingFileLastModifiedDateUTC` in `CompilerDate` from `DateTime` to `string`. 
2. Default value in `CompilerDataLogger` is now `string.Empty`
3. In `BA4001.ReportPortableExecutableCompilerData`, I put the read of the of the last modified date of the pdb in try-catch (since exceptions are possible). If no exceptions, then we get the actual modified date. If exceptions, we emit the empty string (so we let the user care about the interpretation of this).

Can you point me to test binaries on which I can test BA4002?